models/oauth2_application.go

```diff
@@ -559,8 +559,8 @@ func ParseOAuth2Token(jwtToken string) (*OAuth2Token, error) {
 // SignToken signs the token with the JWT secret
 func (token *OAuth2Token) SignToken() (string, error) {
 	token.IssuedAt = time.Now().Unix()
-	jwtToken := jwt.NewWithClaims(jwt.SigningMethodHS512, token)
-	return jwtToken.SignedString(setting.OAuth2.JWTSecretBytes)
+	jwtToken := jwt.NewWithClaims(oauth2.DefaultSigningKey.SigningMethod(), token)
+	return jwtToken.SignedString(oauth2.DefaultSigningKey.SignKey())
```

Comment:
Attempting to verify the generated token using JWT.io, it is unable to parse the signature. It works w/ HS256, but not RS256. I suspect it is something upstream in github.com/dgrijalva/jwt-go that is signing it differently than other libraries expect. I've tried to dig down into the code changes from this PR, but it looks like everything you are setting is correct.

Please let me know if you require any additional information from me to replicate.

Appendix:
You can also see at https://replit.com/@techknowlogick/BruisedCompetitiveLightweightprocess attempting to parse the key, that in headers it is unable to return KeyID (using smallstep CLI to fetch example token from google `step oauth --oidc --bare`, you can see that KeyID is populated).
Sample RS256 token:
```
eyJhbGciOiJSUzI1NiIsInR5cCI6IkpXVCJ9.eyJhdWQiOiI5M2NiNDM5OS04Mzc4LTQ5NGYtYTQwNy1mYzVjODIzYTJjOTIiLCJleHAiOjE2MjI0NDUxMjcsImlhdCI6MTYyMjQ0MTUyNywiaXNzIjoiaHR0cHM6Ly90a2xrLWlubGV0cy5oZXJva3VhcHAuY29tLyIsInN1YiI6IjEiLCJub25jZSI6IjhmZWFlYmYxMzg0MzIwZDNmOTU4YTMxODE0MTQ1NWI5NDk2ZTg4NDA1NTNlZjk1NmVjODcyNGJiYTMzYzQ5MmIifQ.HfksMJPGgjhEP9UJqPulwPvh02VPTgmNkgAxGUdIn8PKiWbPcCkfT2yedFAMccduCqg3FQxLT_LEhPB0T8zuVl4wXdPS-t_qfGem1KE3sZubVdzy2PAA3Iya76kokbk5ufcfnt1bYpGp0p8sCGDm7oOma3N2yvsrXIWIC7EIzPzVICdD7a89ZFMC1TVw2zK_MKU9pUFYcuTrSfrb_Sx5OzpT3DxSMFJ0qfrVIdyYGlwbsujdxaBN2d1BcHv94Ur68FBxxbXoBqggIL7_gOZHmR8ENbRD1D8yJTTqk0pCjR42CqxaosEIv-vqFpLt92J_ZJCr4ldvA0EuVO9HhdjcaDPHrVVW4eh__QOI2aLLTuOcnM165LN5uGt5QN13Y9Ws6rU6inmhycPEqvdK11-o25Y3KlqwgKxNhJLyX-Ctr1Va1y5VZNNK6Oa30bF-_Mb4T92qELMoXthHi5d-wpiiwgURO2xfltxoBrfop0cA7-eCbQ2PPgYZ8oxkXCVGLtlwbcTQXEtzhpu7pz_YBptVtZtnGpqtQqXLcA_afDWuK7KE8xI3vgflSErFwx_QTPF8J_CPnf21IxXp5AoFNUcfZ-k0g7YxnpbX1tNctYGisieTIbW2Jp5iGODdAEKTzsb638jzrz3sOsE1ty01_IrpMkyfHbcbJYeA5SWiL1AA9l8
```

Reply:
JWT.io works for me without changes. Did you change the "Algorithm" on top?
![grafik](https://user-images.githubusercontent.com/1666336/120169415-37fc4400-c200-11eb-884d-4d8b6332a2f7.png)

I used this container to get the token:
```
version: '3.5'

services:
  oidc-test-client:
    image: beryju/oidc-test-client
    ports:
      - 9009:9009
    network_mode: bridge
    environment:
      OIDC_CLIENT_ID: id
      OIDC_CLIENT_SECRET: secret
      OIDC_PROVIDER: gitea-endpoint
      OIDC_DO_INTROSPECTION: "false"
```
I then used
`openssl rsa -in private.pem -out rsa_priv.pem`
and
`openssl rsa -in rsa_priv.pem -pubout > rsa_pub.pem`
to get the private and public key in the desired format.